stl/inc/stacktrace

```diff
@@ -300,16 +273,52 @@ public:
     }
 
 private:
-    static constexpr size_t _Max_frames = 0xFFFF;
+    // <NOTICE> Currently vector is enhanced as unconditionally noexcept move-assignable and swappable
 
     static constexpr bool _Noex_move = allocator_traits<_Alloc>::propagate_on_container_move_assignment::value
-                                    || allocator_traits<_Alloc>::is_always_equal::value;
+                                    || allocator_traits<_Alloc>::is_always_equal::value
+                                    || is_nothrow_move_assignable_v<_Frames_t>; // strengthened
+
+    // <NOTICE> ?_Try_frames may be subject to further decreasements?
+
+    static constexpr size_type _Try_frames = 128; // Sufficient in most cases.
+    static constexpr size_type _Max_frames = 0xFFFF;
 
-    struct _Internal_t {
-        explicit _Internal_t() noexcept = default;
-    };
+    // __declspec(noinline) to make the same behavior for debug and release.
+    // We force the current function to be always noinline and add its frame to skipped.
+    _NODISCARD __declspec(noinline) static basic_stacktrace
+        _Current(unsigned long _Skip, const size_type _Max_depth, const allocator_type& _Al) noexcept {
+        _Skip = _Adjust_skip(_Skip); // skip this frame
+        _TRY_BEGIN
+        basic_stacktrace _Result(_Al);
+
+        stacktrace_entry _Buffer[_Try_frames];
+        unsigned short _Caught =
+            __std_stacktrace_capture(_Skip, static_cast<unsigned long>((_STD min)(_Max_depth, _Try_frames)),
+                reinterpret_cast<void**>(_Buffer), &_Result._Hash);
+        if (_Caught != _Try_frames) { // actual size
+            _Result._Frames.assign(_Buffer, _Buffer + _Caught);
+            return _Result;
+        }
 
-    basic_stacktrace(_Internal_t, size_type _Max_depth, const allocator_type& _Al) : _Frames(_Max_depth, _Al) {}
+        // <NOTICE> The cost introduced by the first try should be compensable from uninitialized resize.
+        // <NOTICE> No need to do _Tidy; ?add _STL_ASSERT for emptiness?
+
+        size_type _Enough_capacity = (_STD min)(_Max_depth, _Max_frames);
+        // resize(_Enough_capacity), but without value initialization
+        // static_assert(is_implicit_lifetime_v<stacktrace_entry>); TRANSITION, not implemented
+        const auto _Newvec = _Result._Frames._Getal().allocate(_Enough_capacity);
+        _Result._Frames._Change_array(_Newvec, _Enough_capacity, _Enough_capacity);
+
+        _Caught = __std_stacktrace_capture(
+            _Skip, static_cast<unsigned long>(_Enough_capacity), _Result._To_voidptr_array(), &_Result._Hash);
+        _Result._Frames.resize(_Caught);
+        return _Result;
+
+        _CATCH_ALL
+        return basic_stacktrace{_Al};
+        _CATCH_END
+    }

```

Comment:
There is a contending implementation with this:
https://github.com/achabense/STL/blob/0277d47749ce8b0fc61ee0863f64b39a51c81ea1/stl/inc/stacktrace#L285-L317

Reply:
Also I'm getting sporadic failures(0 caught size) in BM_bef cases. I'm not sure whether this is a defect in my benchmarking or a problem with `__std_stacktrace_capture`...

<details>
<summary>Test</summary>

```cpp
#define _XKEYCHECK_H
#define private public
#include <stacktrace>
#undef private
#undef _XKEYCHECK_H
//"trace.h" must come first.
#include <benchmark/benchmark.h>

#include <iostream>

using namespace std;

// current std impl
__declspec(noinline) stacktrace
    current_bef(const stacktrace::allocator_type& _Al =
                    stacktrace::allocator_type()) noexcept {
  try {
    stacktrace _Result{stacktrace::_Internal_t{}, stacktrace::_Max_frames, _Al};
    const unsigned short _Actual_size = __std_stacktrace_capture(
        1, static_cast<unsigned long>(stacktrace::_Max_frames),
        _Result._To_voidptr_array(), &_Result._Hash);
    _Result._Frames.resize(_Actual_size);
    return _Result;
  } catch (exception& err) {
    // try to catch the error why result in empty _Result...
    // it turns out that neither std::exception nor other expections will be throw, but the problem still can occur
    cerr << "error:" << err.what() << endl;
    return basic_stacktrace{_Al};
  }
}

#pragma optimize("", off)
template <stacktrace (*current)(const stacktrace::allocator_type&)>
stacktrace get_current(int extra_depth) {
  if (extra_depth > 0) {
    return get_current<current>(extra_depth - 1);
  } else {
    return current({});
  }
}
#pragma optimize("", on)

void BM_bef(benchmark::State& state) {
  const int extra_depth = state.range(0);
  for (auto _ : state) {
    stacktrace cur = get_current<current_bef>(extra_depth);
    if (cur.size() < extra_depth) {
      cout << extra_depth << "->" << cur.size() << endl;
      terminate();
    }
  }
}

BENCHMARK(BM_bef)->Arg(0)->Arg(100)->Arg(150)->Arg(300);

BENCHMARK_MAIN();
```
</details>


![image](https://github.com/microsoft/STL/assets/60953653/b3207e93-524e-47a1-9f14-87d53d77fb40)